### PR TITLE
implemented addCollaborator and removeCollaborator to Repository

### PR DESCRIPTION
**Pull request by marwan-at-work**

### CODE REVIEW

File: lib/Repository.js [hunk 1]
```diff
@@ -368,41 +371,6 @@ class Repository extends Requestable {
...
-    * @param {string} commitSHA - the SHA of the commit that should be updated
-    * @param {object} options - Commit status parameters
-    * @param {string} options.state - The state of the status. Can be one of: pending, success, error, or failure.
-    * @param {string} [options.target_url] - The target URL to associate with this status.
-    * @param {string} [options.description] - A short description of the status.
-    * @param {string} [options.context] - A string label to differentiate this status among CI systems.
-    * @param {Requestable.callback} cb - will receive the updated commit back
-    * @return {Promise} - the promise for the http request
-    */
-   updateStatus(commitSHA, options, cb) {
```
Comment (nmoya) [inline lib/Repository.js:382]: Why remove this function?

File: lib/Repository.js [hunk 2]
```diff
@@ -368,41 +371,6 @@ class Repository extends Requestable {
...
-     * @param {string} [options.homepage] - A URL with more information about the repository
-     * @param {boolean} [options.private] - Either true to make the repository private, or false to make it public.
-     * @param {boolean} [options.has_issues] - Either true to enable issues for this repository, false to disable them.
-     * @param {boolean} [options.has_wiki] - Either true to enable the wiki for this repository, false to disable it.
-     * @param {boolean} [options.has_downloads] - Either true to enable downloads, false to disable them.
-     * @param {string} [options.default_branch] - Updates the default branch for this repository.
-     * @param {Requestable.callback} cb - will receive the updated repository back
-     * @return {Promise} - the promise for the http request
-     */
-   updateRepository(options, cb) {
```
Comment (nmoya) [inline lib/Repository.js:401]: And this?

File: lib/Repository.js [hunk 3]
```diff
@@ -539,7 +534,7 @@ class Repository extends Requestable {
     * @return {Promise} - the promise for the http request
     */
    updatePullRequst(number, options, cb) {
-      log('Deprecated: This method will be removed in next major version. Use updatePullRequest() instead.');
+      log('Deprecated: This method contains a typo and it has been deprecated. It will be removed in next major version. Use updatePullRequest() instead.'); // eslint-disable-line
```
Comment (nmoya) [inline lib/Repository.js]: revert this?

**Review by nmoya — CHANGES_REQUESTED** [review]
The 4th commit is deleting the functions that were implemented in commits 1 and 3.

File: lib/Repository.js (hunk 1 above)
Comment (marwan-at-work) [reply lib/Repository.js:382]: @nmoya i think this happened when rebasing cuz i worked on it on `master` then moved it to the other branch. i'll fix and resubmit. thanks!

**Comment by marwan-at-work** [comment]
@nmoya fixed

File: lib/Repository.js [hunk 4]
```diff
@@ -445,6 +448,33 @@ class Repository extends Requestable {
    }
 
    /**
+    * Add collaborator to this repo.
+    * @see https://developer.github.com/v3/repos/collaborators/#add-user-as-a-collaborator
+    * @param {Object} options - options with username as collaborator - permission (optional)
+    * @param {Requestable.callback} [cb] - will receive the list of PRs
```
Comment (nmoya) [inline lib/Repository.js]: List of PRs? isn't the repositories, the invitee and the inviter?

File: lib/Repository.js [hunk 5]
```diff
@@ -445,6 +448,33 @@ class Repository extends Requestable {
...
+      const payload = options.permission ? {permission: options.permission} : null;
+
+      return this._request('PUT', `/repos/${this.__fullname}/collaborators/${options.username}`, payload, cb);
+   }
+
+   /**
+    * Remove collaborator to this repo.
+    * @see https://developer.github.com/v3/repos/collaborators/#remove-user-as-a-collaborator
+    * @param {Object} options - options with username as collaborator
+    * @param {Requestable.callback} [cb] - will receive the list of PRs
```
Comment (nmoya) [inline lib/Repository.js]: same

File: lib/Repository.js [hunk 6]
```diff
@@ -445,6 +448,33 @@ class Repository extends Requestable {
    }
 
    /**
+    * Add collaborator to this repo.
+    * @see https://developer.github.com/v3/repos/collaborators/#add-user-as-a-collaborator
+    * @param {Object} options - options with username as collaborator - permission (optional)
```
Comment (nmoya) [inline lib/Repository.js:453]: Can you specify the keys inside the options object like this:

```
    * @param {string} [options.target_url] - The target URL to associate with this status.
    * @param {string} [options.description] - A short description of the status.
    * @param {string} [options.context] - A string label to differentiate this status among CI systems.
```

**Comment by nmoya** [comment]
Other than minors on comments, LGTM.

**Review by nmoya — APPROVED** [review]

File: lib/Repository.js (hunk 4 above)
Comment (marwan-at-work) [reply lib/Repository.js]: @nmoya every single method has this param in the Repository class. They get past to the `_request` method. So I'm not sure what the reason is.

**Pull request merged**